### DISCUSSION TITLE
🚀 Feature request: "This system setting is managed in your config file" type functionality

### DISCUSSION THREAD

**Discussion by dougblackjr** [ExpressionEngine/ExpressionEngine#4177] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
I have a larger site with a load of custom global variables, some managed in the database, some managed in config files, some brought in via $_ENV. This includes upload paths.

Sometimes, I just don't know where this is coming from.

**Describe the solution you'd like**
It would be awesome to be able to get a flag, or some sort of text prompt, on an EE setting or upload pref, to know that a variable is being set in the config that is overriding this. Would greatly help track something down that you're trying to edit or debug.

**Describe alternatives you've considered**
Not sure of alternatives, other than just looking and knowing where things are at. But, in a multi-config site, even having a small flag to say "Config file, not database" would be helpful!

- [X] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
This is interesting. We're about to extend EE own logging capabilities, do you think this could be something like log record "config value xxx is loaded from zzz"? I'm not sure that is possible though:)

**Comment by dougblackjr** [comment]
The log idea isn't bad. I was thinking more of a visual cue in the control panel.

For example, we're adding a CDN to a huge site with multiple upload destinations. I tried changing the URL in the Upload Destination, but it is being managed in the config. We have a few things like that, including in the config, that I just need to be able to see where  it is reading from.

Definitely agree on whether it's possible; was trying to think through how those might get loaded or triggered, but not sure.

**Reply by intoeetive** [reply]
Oh, so more like when it's changed and there's a conflict. 

We could also consider config debug utility, similar to other debu utilities we have in Tools section

**Reply by dougblackjr** [reply]
Not even really that, more of a visual cue whether a variable is set in the database or config file/env. Similar to how Craft tells you it is possible, but actually sharing where the data is coming from.

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/6020323/cc3a6fbb-75a5-45d8-87ee-cefb86fe526b)

**Reply by intoeetive** [reply]
Due to how config is loaded, we don't really know that information. Everything is loaded at the start. But, should be possible to store items that are coming from config file somewhere else and use that info to compare. Just need to decide which solution would work the best and serve most cases

**Reply by bryannielsen** [reply]
I like this idea, I wonder if we could track the 'source' or some 'context' for config items as they're loaded/set in the Config Service
